### PR TITLE
Feature/cook a recipe

### PR DESCRIPTION
**Pull request by DrewBradley**
### Is this a fix or a feature?
Feature and refactor

### What is the change/fix?
Got the "canICookThis" method to function as expected. We will need to refactor and DRY the code as it is currently very long.

### Where should the reviewer start?
Lines 39-63 of user.js in the src file.

### How should this be tested?
Open the project by running ```npm start``` in the project repo and running ```npm start```  in the API repo. And click on the add button on a recipe. The shopping list should print to the console.

### CODE REVIEW

File: src/scripts.js [hunk 1]
```diff
@@ -22,15 +23,9 @@ homeButton.addEventListener('click', cardButtonConditionals);
...
 
-
-
 window.onload = onStartup();
 
 function onStartup() {
-  // getData('http://localhost:3001/api/v1/users')
-  getData("http://localhost:3001/api/v1/ingredients")
-  getData("http://localhost:3001/api/v1/recipes")
-
```
Comment (colefiscus) [inline src/scripts.js:33]: This is what I was talking about re: the GET calls.

File: src/user.js [hunk 2]
```diff
@@ -26,13 +29,75 @@ class User {
...
+    let recipeIngredientsInPantry = []
+    let shoppingList = []
+    recipeIngredients.forEach(recipeIngredient => {
+        let match = pantryIngredients.find(pantryIng => pantryIng.ingredient === recipeIngredient.id)
+        if (match) {
+        recipeIngredientsInPantry.push(match)
+        } else if (!match) {
+        shoppingList.push(`{name: ${recipeIngredient.name}, amount: ${recipeIngredient.quantity.amount}, unit: ${recipeIngredient.quantity.unit}}`)
+      }
+    });
```
Comment (colefiscus) [inline src/user.js:51]: Nice job figuring that out!!

File: src/user.js [hunk 3]
```diff
@@ -26,13 +29,75 @@ class User {
...
+    recipeIngredientsInPantry.forEach(recipeItem => {
+      if (recipeItem.ingredient === recipeIngredients.id) {
+        if (recipeItem.amount >= recipeIngredients.quantity.amount) {
+          return
+        } else if (recipeItem.amount < recipeIngredients.quantity.amount) {
```
Comment (colefiscus) [inline src/user.js:56]: I think we can change this just to an else statement.

**Review by colefiscus — APPROVED** [review]
Excellent work! I think the comments are already gone too?

File: src/scripts.js (hunk 1 above)
Comment (elsafluss) [reply src/scripts.js:33]: I think we don't need them here.

File: src/scripts.js [hunk 4]
```diff
@@ -53,21 +48,17 @@ function viewFavorites() {
     favButton.innerHTML = 'Refresh Favorites'
     cardArea.innerHTML = '';
     user.favoriteRecipes.forEach(recipe => {
-      cardArea.insertAdjacentHTML('afterbegin', `<div id='${recipe.id}'
-      class='card'>
-      <header id='${recipe.id}' class='card-header'>
-      <label for='add-button' class='hidden'>Click to add recipe</label>
-      <button id='${recipe.id}' aria-label='add-button' class='add-button card-button'>
-      <img id='${recipe.id}' class='add'
```
Comment (elsafluss) [inline src/scripts.js:61]: These are mostly formatting changes, looks like

File: src/scripts.js [hunk 5]
```diff
@@ -154,23 +159,26 @@ function populateCards(recipes) {
...
           <label for='add-button' class='hidden'>Click to add recipe</label>
           <button id='${recipe.id}' aria-label='add-button' class='add-button card-button'>
-            <img id='${recipe.id} favorite' class='add'
+            <img id='${recipe.id} favorite' class='add add-button'
             src='https://image.flaticon.com/icons/svg/32/32339.svg' alt='Add to
             recipes to cook'>
           </button>
           <label for='favorite-button' class='hidden'>Click to favorite recipe
           </label>
-          <button id='${recipe.id}' aria-label='favorite-button' class='favorite favorite${recipe.id} card-button'></button>
```
Comment (elsafluss) [inline src/scripts.js:168]: This was changed because I needed to be able to grab that recipe.id

File: src/user.js (hunk 2 above)
Comment (elsafluss) [reply src/user.js:51]: it was not fun until we got it

File: src/user.js [hunk 6]
```diff
@@ -26,13 +29,75 @@ class User {
...
+          let diff = (recipeIngredients.quantity.amount - recipeItem.amount)
+          shoppingList.push(`{name: ${recipeItem.name}, amount: ${diff}, unit: ${recipeIngredient.quantity.unit}}`)
+        }
+      }
+    })
+    console.log(shoppingList);
+  }
+  
+  // canICookThis(recipe) {
```
Comment (elsafluss) [inline src/user.js:65]: I've removed these comments in another PR

**Review by elsafluss — APPROVED** [review]

**Pull request merged**